spec/lib/otis/log_importer_spec.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+RSpec.describe Otis::LogImporter do
+  let(:importer) { described_class.new }
+  let(:fixtures_dir) {
+    ["spec", "fixtures", "ulib-logs", "archive", "macc-ht-web-189.umdl.umich.edu", "var", "log", "babel"]
+  }
+  let(:text_log) { Rails.root.join(*fixtures_dir, "access-imgsrv_downloads.log-20250901") }
+  let(:gzip_log) { Rails.root.join(*fixtures_dir, "access-imgsrv_downloads.log-20250902.gz") }
+
+  around(:each) do |example|
+    HTSSDProxyReport.delete_all
+    # Isolate last import timestamp file from subsequent tests
+    Dir.mktmpdir("otis-log-importer") do |tmpdir|
+      ClimateControl.modify(OTIS_LAST_LOG_IMPORT_PATH: tmpdir) do
+        example.run
+      end
+    end
+  end
+
+  describe "#run" do
+    it "populates the table with two entries" do
+      expect do
+        importer.run
+      end.to change { HTSSDProxyReport.count }.by(2)
+    end
+
+    it "ignores entries earlier than the last run file" do
+      # Write a last run file with the current date
+      File.open(importer.last_log_import_path, "w") do |file|
+        file.puts Time.now
+      end
+      # ... and travel into the future
+      Timecop.travel(Time.now + 7.days) do
+        expect do
+          importer.run
+        end.to change { HTSSDProxyReport.count }.by(0)
```

Comment:
FIXME: This is not a good test. Should keep stats on the number of files read and/or the number of entries inserted, since the uniqueness constraints AND the date math can both keep the count from changing. The test should observe the effects most closely related to the time travel.